pkg/services/frontend/context_middleware.go

```diff
@@ -0,0 +1,49 @@
+package frontend
+
+import (
+	"context"
+	"net/http"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/contexthandler/ctxkey"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+// Minimal copy of contextHandler.Middleware for frontend-service
+// frontend-service doesn't handle authentication or know what signed in users are
+func (s *frontendService) contextMiddleware(tracer tracing.Tracer) web.Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			// Don't modify context so that the middleware span doesn't get propagated as a parent elsewhere
+			_, span := tracer.Start(ctx, "Context middleware")
```

Comment:
I know this is copied so I'm not implying this is new or changed functionality, but, I'm confused why we're not modifying the context here, but then modifying it on ln36 anyway, with what _appears_ to have the same effect...